app/controllers/friendships_controller.rb

```diff
@@ -0,0 +1,11 @@
+class FriendshipsController < ApplicationController
+  def destroy
```

Comment:
Requirements from the project logic, the only way to create friendship is using friend requests, so it's possible to implement it through the FriendshipController but it's not good. So, I tried to improve the quality of friendship creation(created establish and destroy method inside the friendship model). But if you have any idea, I'm open to discuss it. For now, I don't have any other improvements.

Reply:
I must, suggest, though we stick to a simpler implementation which is to have a create action here. We can keep the `friendships` and `friend_request` records inherently independent of each other. The logic should simply be:

- Upon friendship creation (`create` action here) a callback in the `friendship.rb` model will do the necessary updating of the associated `friend_requests` record (i.e, setting it to `accepted`). 

While at times it might be easy to formulate logic it's much harder to follow and debug it should things not work as expected. Sticking to our good ol CRUD methods will make our lives easier and also help others collaborate easily with us. We want our code to be telling a simple story when read by others.